lib/hammer_cli_katello/content_view_version.rb

```diff
@@ -259,6 +259,130 @@ class ExportCommand < HammerCLIKatello::SingleResourceCommand
       end
     end
 
+    class ExportCommand < HammerCLIKatello::SingleResourceCommand
+      include HammerCLIForemanTasks::Async
+
+      action :export
+      command_name "export"
+
+      success_message _("Content view export is available in %{directory}.")
+      failure_message _("Could not export the content view")
+
+      option "--id", "ID", _("Content View Version numeric identifier")
+      option(
+        '--export-dir',
+        'EXPORT_DIR',
+        _("Directory to put content view version export into.")
+      )
+
+      validate_options do
+        option(:option_export_dir).required
+      end
+
+      build_options
+
+      class ContentViewIdParamSource
+        def initialize(command)
+          @command = command
+        end
+
+        def get_options(_defined_options, result)
+          if result['option_id'].nil? && result['option_name']
+            result['option_id'] = @command.resolver.content_view_id(result)
+          end
+          result
+        end
+      end
+
+      def option_sources
+        sources = super
+        idx = sources.index { |s| s.class == HammerCLIForeman::OptionSources::IdParams }
+        sources.insert(idx, ContentViewIdParamSource.new(self))
+        sources
+      end
+
+      def execute
+        cvv = content_view_version
+        repositories = cvv['repositories'].collect do |repo|
+          repository(repo['id'])
+        end
+
+        repositories.each do |repo|
+          repo['packages'] = packages(repo['id'])
+          repo['errata'] = errata(repo['id'])
+        end
+
+        json = export_json(cvv, repositories)
+        create_tar(cvv, repositories, json)
+      end
+
+      def create_tar(cvv, repositories, json)
+        export_file = "export-#{cvv['id']}.json"
+        export_repos_tar = "exported-repos-#{cvv['id']}.tar"
+        export_tar = "export-#{cvv['id']}.tar"
+
+        Dir.chdir("/var/lib/pulp/published/yum/https/repos/") do
```

Comment:
Hardcoded paths always trigger me. If this shouldn't be configurable, at least make it a constant please :)